app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,34 @@
+class QueueItemsController < ApplicationController
+  before_action :require_user
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    video = Video.find(params[:video_id])
+
+    add_item_to_queue(video) unless item_is_already_queued?(video)
+    redirect_to my_queue_path
+  end
+
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    remove_item_from_queue(queue_item)
+    redirect_to my_queue_path
+  end
+
+  private
+
+  def add_item_to_queue(video)
+    QueueItem.create!(video_id: params[:video_id], user: current_user, position: current_user.queue_items.count + 1)
+  end
+
+  def item_is_already_queued?(video)
+    !!QueueItem.where("video_id = ? and user_id = ?", params[:video_id], current_user.id).first

```

Comment:
That is OK, a method with `?` should return either `true` or `false`. However it will work with `nil` as well, because conditional is looking at falsey expression and in Ruby, `false` and `nil` are falsey and everything else is truthy.